search/inmemstor.go

```diff
@@ -0,0 +1,87 @@
+package search
```

Comment:
This approach makes search functionality only depends on inmemory store.  Make search functionality to work on any of the storage options. we can move inmemorystore to storage and allow search functionality can be supported on any storage driver